src/device-state.coffee

```diff
@@ -594,11 +593,23 @@ module.exports = class DeviceState extends EventEmitter
 					.then ([ currentState, targetState ]) =>
 						@deviceConfig.getRequiredSteps(currentState, targetState)
 						.then (deviceConfigSteps) =>
-							if !_.isEmpty(deviceConfigSteps)
-								return deviceConfigSteps
+							noConfigSteps = _.every(deviceConfigSteps, ({ action }) -> action is 'noop')
+							if !_.isEmpty(deviceConfigSteps) && not noConfigSteps
+								return [true, deviceConfigSteps]
```

Comment:
It's not gonna have any real effect, but shouldn't this be `[false, deviceConfigSteps]`?